src/models/version.rs

```diff
@@ -80,58 +80,26 @@ impl Version {
 #[derive(Insertable, Debug, Builder)]
 #[diesel(table_name = versions, check_for_backend(diesel::pg::Pg))]
 pub struct NewVersion<'a> {
+    #[builder(start_fn)]
     crate_id: i32,
+    #[builder(start_fn)]
     num: &'a str,
-    #[builder(default)]
     created_at: Option<&'a NaiveDateTime>,
-    #[builder(default, setter(strip_option))]
     yanked: Option<bool>,
-    #[builder(
-        default = "serde_json::Value::Object(Default::default())",
-        setter(custom)
-    )]
+    #[builder(default = serde_json::Value::Object(Default::default()))]
     features: serde_json::Value,
-    #[builder(default)]
     license: Option<&'a str>,
-    #[builder(default, setter(name = "size"))]
+    #[builder(default, name = "size")]
     crate_size: i32,
     published_by: i32,
     checksum: &'a str,
-    #[builder(default)]
     links: Option<&'a str>,
-    #[builder(default)]
     rust_version: Option<&'a str>,
-    #[builder(default, setter(strip_option))]
     pub has_lib: Option<bool>,
-    #[builder(default, setter(strip_option))]
     pub bin_names: Option<&'a [&'a str]>,
 }
 
-impl NewVersionBuilder<'_> {
-    pub fn features(
-        &mut self,
-        features: &BTreeMap<String, Vec<String>>,
-    ) -> serde_json::Result<&mut Self> {
-        self.features = Some(serde_json::to_value(features)?);
-        Ok(self)
-    }
-
```

Comment:
There will be a new attribute in `3.0.` that would allow expressing this setter as this (apart from defning the custom method on the builder directly):
```rust
#[builder(with = |features: &BTreeMap<...>| -> serde_json::Result<_> { 
  serde_json::to_value(features) 
})]
features: serde_json::Value,
```
(yes it's possible to have a fallible setter this way if you annotate the return type of the closure with a type that has `Result` suffix)